added custom events watching button

frozeman opened the pull request:
This PR adds a watch custom events button to custom contracts. This prevents large contracts from retrieving events in the background. Now the user has to specifically enable this when he is on the contract page.

This will increase app speed, while not preventing users from using events if they are aware of the "risk".

<img width="969" alt="screen shot 2016-05-31 at 16 10 58" src="https://cloud.githubusercontent.com/assets/232662/15677411/75a8cf86-274a-11e6-8a09-9b2ba06bf6a9.png">

alexvandesande commented on app/.meteor/versions:
```diff
@@ -33,8 +33,8 @@ ecmascript-runtime@0.2.10
 ethereum:blocks@0.3.1
-ethereum:dapp-styles@0.5.2
-ethereum:elements@0.7.1
+ethereum:dapp-styles@0.5.1
```
Did you publish this version?

alexvandesande commented:
From my tests, all contracts are listing the same events. Are you sure you're filtering the correct contract to watch only the one on the screen?

frozeman commented on app/client/templates/views/account.js:
```diff
@@ -4,11 +4,82 @@ Template Controllers
+@param {Object} newDocument  the account object with .jsonInterface
+*/
+var addLogWatching = function(newDocument){
+    var contractInstance = web3.eth.contract(newDocument.jsonInterface).at(newDocument.address);
```
Thats the place where it loads the contract with the correct address and starts watching on them... in line 27

alexvandesande replied:
I've seen it. So you're saying this is not happening with you? Because it happens to me, but it can be a separate issue..

frozeman replied:
There was an older geth version, which didn't filter reverted logs and therefore one received the ones from other contracts as well. can you try with the latest develop?

alexvandesande replied:
Weird. I still see that issue. @hiddentao or @evertonfraga could you test this PR and see if you have the same issue?

hiddentao replied:
Yep, I'm seeing this too. Too different contracts showing the same events.

hiddentao commented:
Can you rebase this branch please against develop please?

hiddentao commented:
Even though I'm not watching contract events my app starts slowing and the UI becomes unresponsive.

alexvandesande commented on app/i18n/wallet.en.i18n.json:
```diff
@@ -304,6 +304,8 @@
                 "showContractInfo": "Show contract info",
                 "hideContractInfo": "Hide contract info",
+                "watchContractEvents": "Watch contract events",
+                "stopWatchingContractEvents": "Stop watching contract events",
```
I think a better action would be "Load Latest Events" or simply "Load Events"/"Stop loading events". Since the button states does not persists after you leave the page, and having a wording that seems to indicate this is a state change is misleading. Of course, the app won't stop loading the events while you are there, but then there is also a new block incoming so it would make sense for it not to.

alexvandesande commented:
Fabian, please take a look at my comments and rebase this

alexvandesande commented:
Depends on https://github.com/ethereum/mist/pull/796

frozeman commented:
I fixed a race condition in the custom events, if one gets removed. Also added the removal of transactions, which got removed. This needs to be manually tested, by sending transactions to wallet contracts and see if they arrive.

alexvandesande commented:
Coupled with https://github.com/ethereum/mist/pull/796, now it finally works as is should. I still see some issues on the DAO, when I click the checkbox immediately it clears the previous events, is this intended?
